### PR TITLE
Add support for multiple audio tracks

### PR DESCRIPTION
**Pull request by kiddkaffeine**

### CODE REVIEW

File: theorafile.c [hunk 1]
```diff
@@ -127,15 +128,18 @@ int tf_open_callbacks(void *datasource, OggTheora_File *file, tf_callbacks io)
...
-	vorbis_info_init(&file->vinfo);
-	vorbis_comment_init(&file->vcomment);
 	th_info_init(&file->tinfo);
 	th_comment_init(&file->tcomment);
 
 	/* Is there even data for us to read...? */
 	TF_OPEN_ASSERT(INTERNAL_readOggData(file) <= 0)
 
 	/* Read header */
+	vorbis_info vinfo;
```
Comment (flibitijibibo) [inline theorafile.c:138]: The variables have to be at the top, for VS2010. The function calls can go up to where they used to be, to minimize the diff size.

File: theorafile.c [hunk 2]
```diff
@@ -158,13 +162,22 @@ int tf_open_callbacks(void *datasource, OggTheora_File *file, tf_callbacks io)
...
+			file->vinfo = realloc(file->vinfo, (file->vtracks + 1) * sizeof(file->vinfo[0]));
+			file->vcomment = realloc(file->vcomment, (file->vtracks + 1) * sizeof(file->vcomment[0]));
+			file->vstream = realloc(file->vstream, (file->vtracks + 1) * sizeof(file->vstream[0]));
+			file->vinfo[file->vtracks] = vinfo;
+			file->vcomment[file->vtracks] = vcomment;
+			memcpy(&file->vstream[file->vtracks], &filler, sizeof(filler));
+			file->vtracks += 1;
+			file->vpackets += 1;
+
+			vorbis_info_init(&vinfo);
```
Comment (flibitijibibo) [inline theorafile.c:179]: May need a comment like /* Reset this for other possible Vorbis streams */

File: theorafile.c [hunk 3]
```diff
@@ -173,12 +186,15 @@ int tf_open_callbacks(void *datasource, OggTheora_File *file, tf_callbacks io)
...
+	vorbis_comment_clear(&vcomment);
+	vorbis_info_clear(&vinfo);
+
 	/* No audio OR video? */
 	TF_OPEN_ASSERT(!file->tpackets && !file->vpackets)
 
 	/* Apparently there are 2 more theora and 2 more vorbis headers next. */
 	#define TPACKETS (file->tpackets && (file->tpackets < 3))
-	#define VPACKETS (file->vpackets && (file->vpackets < 3))
+	#define VPACKETS (file->vpackets && (file->vpackets < file->vtracks + 2))
```
Comment (flibitijibibo) [inline theorafile.c:197]:
```suggestion
	#define VPACKETS (file->vpackets && (file->vpackets < (file->vtracks + 2)))
```

File: theorafile.c [hunk 4]
```diff
@@ -350,14 +366,20 @@ void tf_close(OggTheora_File *file)
...
-		ogg_stream_clear(&file->vstream);
+		ogg_stream_clear(&file->vstream[file->vtrack]);
 	}
 
 	/* Metadata */
 	th_info_clear(&file->tinfo);
 	th_comment_clear(&file->tcomment);
-	vorbis_comment_clear(&file->vcomment);
-	vorbis_info_clear(&file->vinfo);
+	for (int i = 0; i < file->vtracks; i++)
```
Comment (flibitijibibo) [inline theorafile.c:375]:
```suggestion
	for (i = 0; i < file->vtracks; i += 1)
```

i has to be declared at the top for VS2010.

File: theorafile.c [hunk 5]
```diff
@@ -418,11 +440,25 @@ void tf_audioinfo(OggTheora_File *file, int *channels, int *samplerate)
...
 	if (samplerate != NULL)
 	{
-		*samplerate = file->vinfo.rate;
+		*samplerate = file->vinfo[file->vtrack].rate;
+	}
+}
+
+int tf_setaudiotrack(OggTheora_File *file, int vtrack)
+{
+	// TODO: Flush buffer immediately if playing?
```
Comment (flibitijibibo) [inline theorafile.c:453]:
```suggestion
	/* TODO: Flush buffer immediately if playing? */
```

**Review by flibitijibibo — CHANGES_REQUESTED** [review]
A lot of this is just style stuff... functionally this seems good!

The only question I have is the TODO... what happens if you change tracks mid-stream? Should we add a track changing button to sdl2test?

**Comment by flibitijibibo** [comment]
Opened up https://github.com/kiddkaffeine/Theorafile/pull/1 for the test program. I checked it out and this works great! Once this is cleaned up I'll add this to FNA as an extension to Video.

File: theorafile.c (hunk 1 above)
Comment (kiddkaffeine) [reply theorafile.c]: Didn't realize that was a style an issue in this century.  Updated.

**Pull request merged**